pandas/_libs/tslibs/strptime.pyx

```diff
@@ -232,6 +278,34 @@ def array_strptime(
             else:
                 val = str(val)
 
+            if iso_format:
+                string_to_dts_failed = string_to_dts(
+                    val, &dts, &out_bestunit, &out_local,
+                    &out_tzoffset, False, fmt, exact
+                )
+                if not string_to_dts_failed:
+                    # No error reported by string_to_dts, pick back up
+                    # where we left off
+                    value = npy_datetimestruct_to_datetime(NPY_FR_ns, &dts)
+                    if out_local == 1:
+                        # Store the out_tzoffset in seconds
+                        # since we store the total_seconds of
+                        # dateutil.tz.tzoffset objects
+                        tz = timezone(timedelta(minutes=out_tzoffset))
```

Comment:
in the analogous block in tslib we then adjust value using tz_localize_to_utc.  do we need to do that here?

Reply:
it happens a few levels up, here:

https://github.com/pandas-dev/pandas/blob/ef0eaa45a999d1ec0e1af0db66998e215f87a2dd/pandas/core/tools/datetimes.py#L335-L344

Reply:
makes sense, thanks.  would it be viable to use the same pattern so we can share more code?

Reply:
that would indeed be good, I'll see what I can do